client/lib/upgrades/actions/free-trials.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Internal dependencies
+ */
+import { cartItems, emptyCart, fillInAllCartItemAttributes } from 'lib/cart-values';
+import productsListFactory from 'lib/products-list';
+import { fullCreditsPayment } from 'lib/store-transactions';
+import { submitTransaction } from './checkout';
+
+const productsList = productsListFactory();
+
+function submitFreeTransaction( partialCart, onComplete ) {
```

Comment:
Could we add docblock comments to these two functions please? That would really help to get a better understanding of how they operate. For example it's not clear what a partial cart is or whether a plan is an object or a string.